CRM/Contribute/BAO/Contribution.php

```diff
@@ -2169,24 +2166,14 @@ public static function transitionComponents($params) {
     if ($contributionStatus === 'Completed') {
 
       // only pending contribution related object processed.
-      if ($previousContriStatusId &&
-        !in_array($previousStatus, [
+      if (!in_array($previousStatus, [
           'Pending',
           'Partially paid',
         ])
       ) {
         // this is case when we already processed contribution object.
         return;
       }
-      elseif (!$previousContriStatusId &&
-        !in_array($contributionStatus, [
```

Comment:
ContributionStatus can't ever be in_array('Pending', 'Partially Paid') if it's completed